src/lib/shim/shim_sys.c

```diff
@@ -154,21 +154,36 @@ bool shim_sys_handle_syscall_locally(long syscall_num, long* rv, va_list args) {
         }
     }
 
-    int straceFd = shim_straceFd();
+    int straceFd = shimshmem_getProcessStraceFd(shim_processSharedMem());
 
     if (straceFd >= 0) {
         // TODO: format the time
         uint64_t emulated_time_ms = shim_sys_get_simtime_nanos();
         pid_t tid = shimshmem_getThreadId(shim_threadSharedMem());
 
         bool oldNativeSyscallFlag = shim_swapAllowNativeSyscalls(true);
+
         char buf[100] = {0};
         int len = snprintf(buf, sizeof(buf), "%018ld [tid %d] %s(...) = %ld\n", emulated_time_ms,
                            tid, syscallName, *rv);
         len = MIN(len, sizeof(buf));
-        if (write(straceFd, buf, len) < 0) {
-            warning("Unable to write to strace log");
+
+        int written = 0;
+        while (1) {
+            int write_rv = write(straceFd, buf + written, len - written);
```

Comment:
`write(2)` returns -1 on error, and the specific error in `errno`

Reply:
Oops good catch. Fixed.